src/components/Product.js

```diff
@@ -0,0 +1,175 @@
+import { useEffect, useState } from 'react';
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faDollarSign } from '@fortawesome/fontawesome-free-solid';
+import { Carousel } from "react-responsive-carousel";
+import "react-responsive-carousel/lib/styles/carousel.min.css";
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+
+function Product({cart, setCart, productInfo}) {
+
+    const [product, setProduct] = useState({});
+    const [loaded, setLoaded] = useState(false);
+    const [selectedSize, setSelectedSize] = useState('');
+    const [selectedQuantity, setSelectedQuantity] = useState(1);
+
+    const fetchProductData = (product_id) => {
+      fetch("http://3.86.87.11:8080/getProductDetails/" + product_id)
+        .then(response => {
+          return response.json()
+        })
+        .then(data => {
+          setProduct(data[0]);
+          setLoaded(true);
+        })
+    }
+
+    function getDiscount(newPrice, oldPrice){
+      var diff = oldPrice - newPrice;
```

Comment:
@rahulmeghwal would it be better to use `const` in cases like this?

Reply:
Have updated the code to use const.